src/main.cpp

```diff
@@ -1352,7 +1352,7 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
 
         // Check against previous transactions
         // This is done last to help prevent CPU exhaustion denial-of-service attacks.
-        if (!CheckInputs(tx, state, view, true, STANDARD_SCRIPT_VERIFY_FLAGS, true))
+        if (fRequireStandard && !CheckInputs(tx, state, view, true, STANDARD_SCRIPT_VERIFY_FLAGS, true))
```

Comment:
I'm fairly sure that this is not correct. You certainly shouldn't be skipping the inputs check when standard-ness is not required, it is part of consensus not standardness.